bcdaworker/main_test.go

```diff
@@ -21,7 +21,7 @@ func TestClearTempDirectory(t *testing.T) {
 	assert.NoError(t, err)
 
 	tempFile := filepath.Join(tempDir, "tmp.txt")
-	err = os.WriteFile(tempFile, []byte("test data"), 0644)
+	err = os.WriteFile(tempFile, []byte("test data"), 0600)
```

Comment:
This makes sense, the caller should be root, and it should only need read and write capability as far as I understand.  A bit concerning if it doesnt work though (but smoke tests should pick that up).